newIDE/app/src/BehaviorsEditor/Editors/Physics2Editor/index.js

```diff
@@ -0,0 +1,565 @@
+// @flow
+import * as React from 'react';
+import { Line, Column } from '../../../UI/Grid';
+import Checkbox from 'material-ui/Checkbox';
+import SelectField from 'material-ui/SelectField';
+import MenuItem from 'material-ui/MenuItem';
+import SemiControlledTextField from '../../../UI/SemiControlledTextField';
+import ImagePreview from '../../../ResourcesList/ResourcePreview/ImagePreview';
+import ResourceSelector from '../../../ResourcesList/ResourceSelector';
+import ResourcesLoader from '../../../ResourcesLoader';
+import ShapePreview from './ShapePreview.js';
+import PolygonEditor from './PolygonEditor.js';
+import {
+  type ResourceSource,
+  type ChooseResourceFunction,
+} from '../../../ResourcesList/ResourceSource.flow';
+import { type ResourceExternalEditor } from '../../../ResourcesList/ResourceExternalEditor.flow';
+
+type Props = {|
+  behavior: Object,
+  project: gdProject,
+  resourceSources: Array<ResourceSource>,
+  onChooseResource: ChooseResourceFunction,
+  resourceExternalEditors: Array<ResourceExternalEditor>,
+|};
+
+type State = {|
+  image: string,
+  imageWidth: number,
+  imageHeight: number,
+|};
+
+export default class Physics2Editor extends React.Component<Props, State> {
+  resourcesLoader: typeof ResourcesLoader;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.resourcesLoader = ResourcesLoader;
+
+    this.state = {
+      image: '',
+      imageWidth: 9,
+      imageHeight: 0,
+    };
+  }
+
+  _setImageSize = (width: number, height: number) => {
+    this.setState({
+      imageWidth: width,
+      imageHeight: height,
+    });
+  };
+
+  _isPolygonConvex(properties: Object) {

```

Comment:
nitpicking: properties is a `gdMapStringPropertyDescriptor` (even if this map to Object for now)